pkg/server/commands/reverseexpand/reverse_expand_test.go

```diff
@@ -443,3 +445,124 @@ func TestReverseExpandIgnoresInvalidTuples(t *testing.T) {
 		}
 	}
 }
+
+func TestReverseExpandThrottle(t *testing.T) {
+	t.Cleanup(func() {
+		goleak.VerifyNone(t)
+	})
+
+	model := testutils.MustTransformDSLToProtoWithID(`model
+  schema 1.1
+type user
+type document
+  relations
+	define viewer: [user]`)
+	mockController := gomock.NewController(t)
+	defer mockController.Finish()
+	mockDatastore := mocks.NewMockOpenFGADatastore(mockController)
+
+	t.Run("dispatch_below_threshold_doesnt_call_throttle", func(t *testing.T) {
+		mockThrottler := mocks.NewMockThrottler(mockController)
+		ctx := context.Background()
+		reverseExpandQuery := NewReverseExpandQuery(
+			mockDatastore,
+			typesystem.New(model),
+			WithDispatchThrottlerConfig(throttler.Config{
+				Throttler:    mockThrottler,
+				Enabled:      true,
+				Threshold:    200,
+				MaxThreshold: 200,
+			}),
+		)
+		mockThrottler.EXPECT().Throttle(gomock.Any()).Times(0)
+		dispatchCountValue := uint32(190)
+		metadata := NewResolutionMetadata()
+		metadata.DispatchCount = &dispatchCountValue
+
+		reverseExpandQuery.throttle(ctx, metadata)
+	})
```

Comment:
do we need to check whether metadata.WasThrottled = false?

Reply:
Added this 👍